### PR TITLE
refactor(formula): align to template-formula & fix #19

### LINKED ISSUE #19: Create more storage objects in onetime
Hi, I wonder if there is a proper way to create more storage_objects in onetime. 
I tried some solutions but all failed, I have this pillar file:

```
iscsi:
  isns:
    enabled: false
  target:
    lio:
      myconf:
        fabric_modules:
          authenticate_target: 'false'
          enforce_discovery_auth: 'false'
          name: "iscsi"
        storage_objects:
          attributes:
            block_size: 512
            emulate_write_cache: 0
            queue_depth: 64
            unmap_granularity: 0
          dev: "/dev/xvdd1"
          name: "sda"
          plugin: "block"
        targets:
          fabric: iscsi
          tpgs:
            attributes:
              authentication: 0
              cache_dynamic_acls: 0
              default_cmdsn_depth: 16
              demo_mode_write_protect: 0
              generate_node_acls: 1
              login_timeout: 15
              netif_timeout: 2
              prod_mode_write_protect: 0
            luns:
              index: 0
              storage_object: "/backstores/block/sda"
            portals:
              ip_address: xxxxxxxxx
              port: 3260
            tag: 1
          wwn: "iqn.1996-04.blabla:01:a66aed20e2f3"
```

I would like to add /dev/xvdd2 ==> sdb ==> lun1 and so one.
Maybe I'm missing something here, do you have an idea?

### PR DESCRIPTION
**Pull request by noelmcloughlin**
This PR aligns entire formula to the `template-formula` standardization and fixes the major limitation described in #19 

Verification will be done on FreeBSD and GNU/Linux - results reported in comments below.

### CODE REVIEW

**Comment by noelmcloughlin** [comment]
Verified on CentOS7: Here is output: [saveconfig.json.centos7.txt](https://github.com/saltstack-formulas/iscsi-formula/files/3622484/saveconfig.json.centos7.txt)

**Comment by noelmcloughlin** [comment]
Verified on Ubuntu now: [iscsi.ubuntu.txt](https://github.com/saltstack-formulas/iscsi-formula/files/3623149/iscsi.ubuntu.txt)

The `tgt` service did not start because I tested logical configuration only (no backing storage).

**Comment by noelmcloughlin** [comment]
Verified on OpenSUSE now:  [iscsi-opensuse-salt.output.txt](https://github.com/saltstack-formulas/iscsi-formula/files/3623444/iscsi-opensuse-salt.output.txt)

The `target.service` does not exist - I asked SuSE guys for help: https://github.com/yast/yast-iscsi-lio-server/issues/96
This is good enough for now.

**Comment by noelmcloughlin** [comment]
Now working on OpenSUSE: [iscsi-suse-lio-targetcli.out.txt](https://github.com/saltstack-formulas/iscsi-formula/files/3623517/iscsi-suse-lio-targetcli.out.txt)

**Comment by noelmcloughlin** [comment]
Now working on Archlinux:  [archlinux-iscsi-target-fb.out.txt](https://github.com/saltstack-formulas/iscsi-formula/files/3623705/archlinux-iscsi-target-fb.out.txt)

Notes:
- I ran `users-formula` first to create necessary user.
- Before running iscsi formula I rebooted so kernel upgrade is appllied
-  I found this salt bug ...

`unless: {{ var }}`  evaluated to `unless: https://aur.archlinux.org` which failed.]

The fix I used was:
`unless: {{ var != None }}`

**Comment by noelmcloughlin** [comment]
Verified on FreeBSD 11.2:  [freebsd-istgt-output.txt](https://github.com/saltstack-formulas/iscsi-formula/files/3623746/freebsd-istgt-output.txt)

This PR is fully verified now.

**Comment by myii** [comment]
@noelmcloughlin Getting all failures on Travis at the moment:

* https://travis-ci.org/myii/iscsi-formula/builds/586323276

Specifically:

* https://travis-ci.org/myii/iscsi-formula/jobs/586323277#L1446-L1454

```salt
                 ID: iscsi-target-service-install-service-running
           Function: service.running
               Name: tgt
             Result: False
            Comment: Failed to start tgt.service: Unit tgt.service not found.
            Started: 01:17:00.449964
           Duration: 120.00000000000455 ms
            Changes:   
         Name: iscsi-target-service-install-failure-explanation - Function: test.show_notification - Result: Clean Started: - 01:17:00.572500 Duration: 9.999999999990905 ms
```

Can you enable Travis on your side and test it out?

**Comment by baby-gnu** [comment]
Hello.

The ` docs/_static/css/custom.css`, ` docs/conf.py` and ` docs/index.rst` are deprecated and should not be included (https://github.com/saltstack-formulas/libvirt-formula/pull/43#pullrequestreview-253285365).

I think they should be removed from `template-formula`.

Regards.

**Comment by noelmcloughlin** [comment]
> Can you enable Travis on your side and test it out?

Could we disable Debian/Ubuntu on Travis for the moment? The `tgt` service will not start because the logical configuration has no backing storage (created by lvm formula for example).  It's not something I can easily fix so a separate PR would be better.

**Comment by noelmcloughlin** [comment]
> The ` docs/_static/css/custom.css`, ` docs/conf.py` and ` docs/index.rst` are deprecated and should not be inc

Thanks @baby-gnu for reviewing. I will remove those files from here.

**Comment by myii** [comment]
> Could we disable Debian/Ubuntu on Travis for the moment? The `tgt` service will not start because the logical configuration has no backing storage (created by lvm formula for example). It's not something I can easily fix so a separate PR would be better.

That's not a problem but we should get it passing on some of the other platforms instead.  It's possible to set dependencies on other formulas in `kitchen.yml` like [this example](https://github.com/saltstack-formulas/packages-formula/blob/1aa6f5009149fcdeb757380e9780e0f7662378af/kitchen.yml#L191-L210).  Please try to do the testing in your own account in Travis, so that it doesn't block other formulas that are using Travis at the same time; at the current time, we've only got 3 platforms that can run concurrently across all of the formulas in the whole org.

File: iscsi/osfamilymap.yaml [hunk 1]
```diff
@@ -33,46 +33,63 @@ Debian:
...
       wanted:
-        - libvirt-daemon-driver-storage-iscsi
-        - netbsd-iscsi
-        - udisks2-iscsi
-        - yum-plugin-versionlock
-        - targetcli
-        - device-mapper-persistent-data
-        # scsi-target-utils
-        # fcoe-target-utils
+	- device-mapper-persistent-data                #cent6/cent7/fedora/amazon
```
Comment (n-rodriguez) [inline iscsi/osfamilymap.yaml]: Be careful to use consistent indentation.

Comment (noelmcloughlin) [reply iscsi/osfamilymap.yaml]: Got it - thanks @n-rodriguez

**Comment by noelmcloughlin** [comment]
Travis is 50% working now: https://travis-ci.org/noelmcloughlin/iscsi-formula/builds/586794185

Once https://github.com/saltstack-formulas/lvm-formula/pull/13 is merged I'll try again as this may fix RedHat family (centos, fedora, amazon) otherwise I'll fix travis without using lvm.

**Comment by noelmcloughlin** [comment]
Travis status

- `INSTANCE=default-centos-7-2019-2-py3` 
    mage lacks `target_mod_core` kernel module so **target** service cannot start.

- `INSTANCE=default-centos-6-2017-7-py2`
     `isnsd` service fails with "none of the trigger conditions were met".

- INSTANCE=default-amazonlinux-2-2018-3-py2
     `iscsi` service fails with "none of the trigger conditions were met".

I will drop CentOS6 because technologically it has an obsolete iscsi implementation.
I hope to get CentOS7 & Amazonlinux working

**Comment by noelmcloughlin** [comment]
Hi @myii 
I can get 5 of 6 Travis jobs passing. That is the best I can achieve in this PR.
https://travis-ci.org/noelmcloughlin/iscsi-formula/builds/587598962

What is the next stage?
thanks!!

**Comment by myii** [comment]
@noelmcloughlin Noticed that you're making progress and only one instance failing -- good work.  I ran the `lint` stage as well and found that there are going to be numerous `yamllint` errors:

* https://travis-ci.com/saltstack-formulas/iscsi-formula/jobs/237749889#L222-L279

```bash
$ yamllint -s .
./kitchen.yml
  154:5     error    wrong indentation: expected 6 but found 4  (indentation)
  182:5     error    wrong indentation: expected 6 but found 4  (indentation)
  217:5     error    wrong indentation: expected 6 but found 4  (indentation)

./pillar.example
  639:1     error    too many blank lines (1 > 0)  (empty-lines)
  4:6       error    syntax error: found character '%' that cannot start any token

./iscsi/defaults.yaml
  6:21      warning  missing starting space in comment  (comments)
  13:21     warning  truthy value should be one of [false, true]  (truthy)
  19:12     error    trailing spaces  (trailing-spaces)
  30:16     warning  truthy value should be one of [false, true]  (truthy)
  31:15     warning  truthy value should be one of [false, true]  (truthy)
  35:32     warning  missing starting space in comment  (comments)
  39:32     warning  missing starting space in comment  (comments)
  40:32     warning  missing starting space in comment  (comments)
  41:32     warning  missing starting space in comment  (comments)
  42:32     warning  missing starting space in comment  (comments)
  43:32     warning  missing starting space in comment  (comments)
  44:32     warning  missing starting space in comment  (comments)
  62:14     error    empty value in block mapping  (empty-values)
  66:14     error    empty value in block mapping  (empty-values)
  69:14     error    empty value in block mapping  (empty-values)
  72:14     error    empty value in block mapping  (empty-values)
  94:14     warning  truthy value should be one of [false, true]  (truthy)
  96:14     warning  truthy value should be one of [false, true]  (truthy)
  105:14    warning  truthy value should be one of [false, true]  (truthy)
  106:17    warning  truthy value should be one of [false, true]  (truthy)
  108:14    warning  truthy value should be one of [false, true]  (truthy)
  118:17    warning  truthy value should be one of [false, true]  (truthy)
  119:14    warning  truthy value should be one of [false, true]  (truthy)
  121:14    warning  truthy value should be one of [false, true]  (truthy)

./iscsi/osfamilymap.yaml
  18:14     warning  truthy value should be one of [false, true]  (truthy)
  18:22     warning  missing starting space in comment  (comments)
  35:25     warning  too few spaces before comment  (comments)
  35:26     warning  missing starting space in comment  (comments)
  36:26     warning  missing starting space in comment  (comments)
  59:17     error    trailing spaces  (trailing-spaces)
  106:8     error    wrong indentation: expected 8 but found 7  (indentation)
  114:17    warning  truthy value should be one of [false, true]  (truthy)
  137:17    warning  truthy value should be one of [false, true]  (truthy)

./iscsi/osfingermap.yaml
  36:17     warning  truthy value should be one of [false, true]  (truthy)
  56:17     warning  truthy value should be one of [false, true]  (truthy)

./iscsi/oscodename.yaml
  4:2       error    syntax error: found character '%' that cannot start any token
  20:89     error    line too long (94 > 88 characters)  (line-length)
  26:89     error    line too long (105 > 88 characters)  (line-length)

./test/salt/pillar/centos6.sls
  4:6       error    syntax error: mapping values are not allowed here
```

When you finally get the tests working, we can discuss how to fix (or ignore) the errors above.

**Comment by noelmcloughlin** [comment]
Thanks @myii I have fixed the lint issues.

Travis has regressed after https://github.com/saltstack-formulas/lvm-formula/pull/14 got merged (I think) so I raised a new PR  https://github.com/saltstack-formulas/lvm-formula/pull/15 

The dependency (lvm) is causing the Travis failures now afaik.

**Comment by noelmcloughlin** [comment]
Travis CI is fixed now.

**Pull request merged**